src/interface/contourplot.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    contourplot(x, y, A; kwargs...)
+
+Draws a contour plot on a new canvas.
+
+# Arguments
+- `levels`: the number of contour levels. 
+- `colormap`: the colormap to use for the contour lines. 
+- `colorbar`: toggle the colorbar. 
+$DOC_PLOT_PARAMS
+- `height`: number of character rows that should be used for plotting.
+- `xlim`: plotting range for the x axis. `(0, 0)` stands for automatic.
+- `ylim`: plotting range for the y axis. `(0, 0)` stands for automatic.
+- `grid`: if `true`, draws grid-lines at the origin.
+
+# Author(s)
+- T Bltg (https://github.com/t-bltg)
+"""
+function contourplot(
+    x::AbstractVector, y::AbstractVector, A::AbstractMatrix;
+    canvas::Type = BrailleCanvas, name::AbstractString = "", levels::Integer = 3, colormap = :viridis,
+    blend = false, grid = false, colorbar = true, kw...
+)   
+    callback = colormap_callback(colormap)
+    plot = Plot(
+        extrema(x) |> collect,
+        extrema(y) |> collect, canvas;
+        blend = blend, grid = grid, colorbar = colorbar, colormap = callback,
+        kw...
+    )
+    contourplot!(plot, x, y, A; name = name, levels = levels, colormap = callback)
+end
+
+function contourplot!(
+    plot::Plot{<:Canvas}, x::AbstractVector, y::AbstractVector, A::AbstractMatrix;
+    name::AbstractString = "", levels::Integer = 3, colormap = :viridis, 
+)
+    name == "" || label!(plot, :r, string(name))
+
+    plot.colormap = callback = colormap_callback(colormap)
+    mA, MA = extrema(A)
+
+    for cl in Contour.levels(Contour.contours(y, x, A, levels))
+        color = callback(Contour.level(cl), mA, MA)
+        for line in Contour.lines(cl)
+            yi, xi = Contour.coordinates(line)
+            lineplot!(plot, xi, yi, color=color)
+        end
+    end
+    plot
+end
+
+"""
+    contourplot(A; kwargs...)
+
+Contour plot for images (flips the y axis by default)
+"""
+contourplot(A::AbstractMatrix; kwargs...) =
+    contourplot(axes(A, 2) |> collect, axes(A, 1) |> collect |> reverse, A; kwargs...)
```

Comment:
How about adding one extra method

```julia
contourplot(A::AbstractMatrix{CT}; kwargs...) where {CT<:Gray} = contourplot(reinterpret(eltype(CT), A); kwargs...)
```

so that `contourplot(testimage("cam"))` just works?

Reply:
Doesn't this require adding `Colors.jl` dependency ? Which lead to another question: should we support `colorant`s as given colors ?

Reply:
Oh, I see... That's a good question. If we're going to introduce ImageInTerminal encoding backend in the future, we'd eventually get those dependencies and have colorants supported...

Reply:
What I propose is to open an issue for feature request, and we can support that later, I'll write the PR, or maybe integrate this change in https://github.com/JuliaPlots/UnicodePlots.jl/pull/202. Is is ok for you ?